pkgs/servers/mautrix-signal/default.nix

```diff
@@ -32,16 +34,30 @@ buildGoModule rec {
     })
   ];
 
-  buildInputs = (lib.optional (!withGoolm) olm) ++ [
-    # must match the version used in https://github.com/mautrix/signal/tree/main/pkg/libsignalgo
-    # see https://github.com/mautrix/signal/issues/401
-    libsignal-ffi
-  ];
+  buildInputs =
+    (lib.optional (!withGoolm) olm)
+    ++ (lib.optional withGoolm stdenv.cc.cc.lib)
+    ++ [
+      # must match the version used in https://github.com/mautrix/signal/tree/main/pkg/libsignalgo
+      # see https://github.com/mautrix/signal/issues/401
+      libsignal-ffi
+    ];
+
   tags = lib.optional withGoolm "goolm";
 
+  CGO_LDFLAGS = [ "-lstdc++" ];
+
   vendorHash = "sha256-bKQKO5RqgMrWq7NyNF1rj2CLp5SeBP80HWxF8MWnZ1U=";
 
-  doCheck = false;
+  doCheck = true;
+  preCheck = ''
+    # Needed by the tests to be able to find libstdc++
+    export LD_LIBRARY_PATH="${stdenv.cc.cc.lib}/lib:$LD_LIBRARY_PATH"
```

Comment:
This might become unnecessary by providing the `-L${stdenv.cc.cc.lib}/lib` flag to `CGO_LDFLAGS` (testing myself in a moment): https://github.com/NixOS/nixpkgs/blob/7ffd9ae656aec493492b44d0ddfb28e79a1ea25d/pkgs/servers/blockbook/default.nix#L47